falcon_mcp/modules/ngsiem.py

```diff
@@ -0,0 +1,805 @@
+"""NG-SIEM module for Falcon MCP Server.
+
+This module provides core functionality for executing LogScale/CQL queries against
+CrowdStrike NG-SIEM with a modular, extensible architecture designed for future enhancements.
+
+The module contains:
+- NGSIEMConfig: Configuration management for query execution parameters
+- NGSIEMQueryEngine: Core query execution engine with hook system for extensibility
+- NGSIEMModule: MCP module for registering tools and handling query requests
+
+Typical usage example:
+
+  config = NGSIEMConfig()
+  module = NGSIEMModule(client, config)
+  result = module.execute_ngsiem_query("#event_simpleName=ProcessRollup2 | tail(3)")
+"""
+
+import time
+from datetime import datetime, timedelta, timezone
+from typing import Any, Callable, Dict, List, Optional
+
+from mcp.server import FastMCP
+from pydantic import Field
+
+from falcon_mcp.common.errors import handle_api_response
+from falcon_mcp.common.logging import get_logger
+from falcon_mcp.common.utils import prepare_api_parameters
+from falcon_mcp.modules.base import BaseModule
+
+logger = get_logger(__name__)
+
+
+class NGSIEMConfig:
+    """Configuration for NGSIEM query engine designed for future extensibility.
+
+    Provides configuration settings for query execution, resource management,
+    and output formatting. Future features like query validation, result caching,
+    and query templates are included but disabled by default.
+
+    Attributes:
+        timeout_seconds: Maximum time to wait for query completion in seconds.
+        polling_interval: Time between status checks in seconds.
+        max_attempts: Maximum number of polling attempts before timeout.
+        enable_query_validation: Whether to enable query syntax validation.
+        enable_result_caching: Whether to enable result caching.
+        enable_query_templates: Whether to enable query templates.
+        enable_data_analysis: Whether to enable data analysis tools.
+        enable_field_search: Whether to enable field search helpers.
+        cleanup_on_timeout: Whether to cleanup searches that timeout.
```

Comment:
I think this two should be a default behaviour as I don't see cases we wouldn't want to clean up the searches. We can keep this simple and always clean up on timeout or error